joomla/joomla-cms
# Restore onChange for calendar

Pull Request for Issue #14306 .

### Summary of Changes
- Restore the onChange property
- Add an extra check that the field belongs to a form before attaching an `on submit` listener

### Testing Instructions
Apply patch
Edit protostar template and insert somewhere inside the body:
```html
<?php echo JHtml::_('calendar', JFactory::getDate()->format('Y-m-d'), 'date', 'date', '%Y-%m-%d', array('onChange' => "alert('it works')")); ?>
```
An alert with a message `it works` should appear each time you change the date in the calendar

### Documentation Changes Required
None, this is a bug
